packages/trader/src/exchange-adapters/UniswapV3Trader.ts

```diff
@@ -0,0 +1,59 @@
+import winston from "winston";
+import Web3 from "web3";
+import BigNumber from "bignumber.js";
+
+const bn = require("bignumber.js"); // Big number that comes with web3 does not support square root.
+bn.config({ EXPONENTIAL_AT: 999999, DECIMAL_PLACES: 40 });
+
+const { encodePriceSqrt } = require("@uma/common");
+const ExchangeAdapterInterface = require("./ExchangeAdapterInterface");
+const { getTruffleContract } = require("@uma/core");
+
+class UniswapV3Trader implements InstanceType<typeof ExchangeAdapterInterface> {
```

Comment:
nit: why do we need to use `InstanceType<typeof ExchangeAdapterInterface>`? My guess is that this is because we're using JS style requires rather than typescript `import X from Y;` syntax.

Reply:
could be. will try to update to the correct syntax and see.

Reply:
ye this is a problem in a few more places. will update it throught.